internal/ui/ui_mobile.go

```diff
@@ -300,6 +305,15 @@ func (m *Monitor) ensureInit() {
 	m.width = width
 	m.height = height
 	m.deviceScaleFactor = scale
+
+	safeAreaLeft, safeAreaTop, safeAreaRight, safeAreaBottom, ok := theUI.safeArea()
+	if !ok {
+		return
```

Comment:
Do not early return here. `m.inited.Store(true)` is skipped

Reply:
This is fragile when we add more logics. Do this

```go
safeAreaLeft, safeAreaTop, safeAreaRight, safeAreaBottom, ok := theUI.safeArea()
if ok {
	m.safeAreaLeft = safeAreaLeft
	m.safeAreaTop = safeAreaTop
	m.safeAreaRight = safeAreaRight
	m.safeAreaBottom = safeAreaBottom
}
```

(Moreover, the four values should be image.Rectangle. I'll comment this later)